flexmeasures/ui/views/analytics.py

```diff
@@ -676,3 +663,21 @@ def revenue_unit_factor(quantity_unit: str, price_unit: str) -> float:
         return 1000
     else:
         raise NotImplementedError
+
+
+def determine_resolution(data: pd.DataFrame) -> str:
+    """
+    Determine the resolution to be displayed under the plot.
+    We try to get it from the DataFrame's meta data, or guess from the actual data.
+    Lastly, we try the session.
+    If nothing can be found this way, the resulting string is "?"
+    """
+    if hasattr(data.index, "freqstr") and data.index.freqstr is not None:
```

Comment:
I suggest using `if hasattr(data, "event_resolution")`, too, making this function suitable for a BeliefsDataFrame, too. Can't hurt.

Reply:
Okay. If it's there, it can actually be a lot of things (`BeliefsDataFrame.event_resolution` is a  `TimedeltaLike`, which is `Union[timedelta, str, pd.Timedelta]`). I know how to turn the first into a freqstr (we have `time_utils.timedelta_to_pandas_freqstr`), but I'm not sure about the `str` and `pd.Timedelta`). Do you know what that would require?

Reply:
Just for reference: timely-beliefs currently guarantees that `event_resolution` is a datetime.timedelta, not a TimedeltaLike (internally, it uses `timely_beliefs.utils.parse_timedelta_like` to convert to a timedelta). If that changes to a TimedeltaLike in the future, timely-beliefs should warn us of the change. In case of a string type, it would already be the freq_str we want. In case of a pd.Timedelta, we'd need `from pandas.tseries.frequencies import to_offset`.